metis-performance-metric/src/main/java/eu/europeana/metis/performance/metric/utilities/CSVUtilities.java

```diff
@@ -0,0 +1,109 @@
+    public CSVUtilities(MongoMetisCoreDao mongoMetisCoreDao) {
+        this.mongoMetisCoreDao = mongoMetisCoreDao;
+    }
+
+    public void writeIntoCsvFile(String filePath, Date startDate, Date endDate) throws FileNotFoundException {
+        final File resultFile = new File(filePath);
+        final String firstRow = "DatasetId, Date Publish Indexing, Time of last successful harvest in hours, Number of Records Published";
+
+        try (PrintWriter printWriter = new PrintWriter(resultFile)) {
+            List<WorkflowExecutionDao.ExecutionDatasetPair> workflowExecutionsOverview =
+                    mongoMetisCoreDao.getAllWorkflowsExecutionsOverview(startDate, endDate).getResults();
+            List<WorkflowExecution> cleanedWorkflowExecutionList = cleanUpWorkflowExecutionList(workflowExecutionsOverview);
+            cleanedWorkflowExecutionList.sort(Comparator.comparing(workflow -> Integer.parseInt(workflow.getDatasetId())));
+            List<String> contentToPrint = cleanedWorkflowExecutionList.stream()
+                    .map(this::turnDatasetIntoCSVRow)
+                    .filter(StringUtils::isNotEmpty)
+                    .collect(Collectors.toList());
+            printWriter.println(firstRow);
+            contentToPrint.forEach(content -> {
+                if(StringUtils.isNotEmpty(content)){
+                    printWriter.println(content);
+                }
+            });
+        }
+
+    }
+
+    private String turnDatasetIntoCSVRow(WorkflowExecution execution){
+        String datasetId = execution.getDatasetId();
+        SimpleDateFormat simpleDateFormatter = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+        LOGGER.info("Processing dataset with id " + datasetId);
+        StringBuilder stringBuilderCSVRow = new StringBuilder();
+        stringBuilderCSVRow.append(datasetId);
+        stringBuilderCSVRow.append(", ");
+        Optional<AbstractMetisPlugin> optionalPublishPlugin = execution.getMetisPluginWithType(PluginType.PUBLISH);
+        if(optionalPublishPlugin.isPresent()){
+            AbstractMetisPlugin publishPlugin = optionalPublishPlugin.get();
+            stringBuilderCSVRow.append(simpleDateFormatter.format(publishPlugin.getFinishedDate()));
+            stringBuilderCSVRow.append(", ");
+            stringBuilderCSVRow.append(calculateTimeDifference(execution, (AbstractExecutablePlugin<?>)publishPlugin));
+            stringBuilderCSVRow.append(", ");
+            stringBuilderCSVRow.append(((AbstractExecutablePlugin<?>) publishPlugin).getExecutionProgress().getProcessedRecords());
+        } else {
+            LOGGER.error("Something went wrong when extracting data");
+            return "";
+        }
+
+        LOGGER.info("Finished processing dataset with id " + datasetId);
+
+        return stringBuilderCSVRow.toString();
+    }
+
+    private long calculateTimeDifference(WorkflowExecution workflowExecution, AbstractExecutablePlugin<?> publishPlugin){
```

Comment:
One more suggestion to give a sense of time in hours using decimals, if a harvesting lasted minutes is displayed '0' zero
using a double will give for example 0.5 hrs
```
private double calculateTimeDifference(WorkflowExecution workflowExecution, AbstractExecutablePlugin<?> publishPlugin){
        PluginWithExecutionId<? extends ExecutablePlugin> harvestPlugin =
                mongoMetisCoreDao.getHarvesting(new PluginWithExecutionId<>(workflowExecution,publishPlugin));
        return TimeUnit.MILLISECONDS.toMinutes(publishPlugin.getFinishedDate().getTime() - harvestPlugin.getPlugin().getStartedDate().getTime()) / 60.0;
    }
```